src/main/java/org/springframework/data/elasticsearch/core/ReactiveElasticsearchTemplate.java

```diff
@@ -894,4 +895,55 @@ private RuntimeException translateException(Throwable throwable) {
 		return Mono.just(entity);
 	}
 	// endregion
+
+	protected interface DocumentCallback<T> {
```

Comment:
first thought was that using the same name as in non-reactive code might be not the best, but as these are protected nested classes, this should be no problem.